doc/nrf/app_dev/device_guides/nrf54l/ecies_x25519.rst

```diff
@@ -0,0 +1,80 @@
+
+While the use of SHA-256 does not pose a security concern and has minimal impact on performance, it increases code size because SHA-256 support must be included alongside SHA-512, which is already used by the ED25519 signature algorithm.
+
+Additionally, pre-installed MCUboot instances will not be able to boot images that use TLVs generated with different hash algorithms.
+
+Building application with image encryption
+******************************************
+
+To build an application that uses MCUboot with image encryption enabled, run the following command:
+
+.. parsed-literal::
+   :class: highlight
+
+    west build -b *board_target* -- -DSB_CONFIG_BOOTLOADER_MCUBOOT=y -DSB_CONFIG_BOOT_ENCRYPTION=y
+
+The :kconfig:option:`SB_CONFIG_BOOT_ENCRYPTION` option enables encryption support in MCUboot.
+
+The key exchange method is determined by the type of signature key selected.
+For the nRF54L15 SoC, the ED25519 signature algorithm is selected by default.
+
+When encryption is enabled, the encrypted image is generated in the application build directory under the names :file:`zephyr.signed.encrypted.bin` and :file:`zephyr.signed.encrypted.hex`.
+
+The BIN file is a binary image suitable for DFU operations using :ref:`MCUmgr<dfu_tools_mcumgr_cli>`.
+
+When the :kconfig:option:`SB_CONFIG_BOOT_ENCRYPTION_KEY_FILE` option is enabled, you must provide an ECIES-X25519 private key in PEM format.
+This key is built into the MCUboot image during the build process.
+See the following example:
+
+.. parsed-literal::
+   :class: highlight
+
+   west build -b *board_target* -- -DSB_CONFIG_BOOTLOADER_MCUBOOT=y -DSB_CONFIG_BOOT_ENCRYPTION=y -DSB_CONFIG_BOOT_ENCRYPTION_KEY_FILE=\"<path to key.pem>\"
+
+.. note::
+   * The public key, derived from the specified private key file, is added to the image and later used on the device to derive the decryption key for the application image.
+     This public key is automatically derived from the private key by ``imgtool``, which is invoked by the build system when signing the image.
+   * The sysbuild option :kconfig:option:`SB_CONFIG_BOOT_ENCRYPTION` sets the MCUboot configuration option :kconfig:option:`CONFIG_BOOT_ENCRYPT_IMAGE`.
+     Similarly, the :kconfig:option:`SB_CONFIG_BOOT_ENCRYPTION_KEY_FILE` option sets both :kconfig:option:`CONFIG_BOOT_ENCRYPTION_KEY_FILE` for MCUboot and :kconfig:option:`CONFIG_MCUBOOT_ENCRYPTION_KEY_FILE` for the default application.
+
+   These values are then passed to ``imgtool`` for encrypting the application image.
+
+   You cannot override these options using MCUboot or application-level Kconfig options, as they are enforced by sysbuild.
+
+Enabling encryption in |nRFVSC| projects
+****************************************
+
+Generally, enabling encryption in projects created using |nRFVSC| requires passing Kconfig options, mentioned in above paragraphs, to project configuration. This action is best described in :ref:`cmake_options`.
```

Comment:
I don't think these are solid instructions that the user could follow. It says "generally, it works like this", which almost sounds like it wasn't checked